File: Challengs/Sum-Number-MaxDigit/SumNumMaxDigit.py
import random
def sum_num_max_digit(max_digit: int):
    num = []
    sum = 0
    length_digit = 4

    for i in range(length_digit):
        digit = random.randint(1, max_digit)
        num.append(digit)
        sum += digit

    total_digit = ''.join(map(str, num))
    
    if sum == 21: return total_digit
    return sum_num_max_digit(max_digit)

print(sum_num_max_digit(6))

File: Challengs/Sum-Number-MaxDigit/SumNumMaxDigit.ts
const SumNumMaxDigit = (maxDigit: number): number => {
  let num = [];
  let sum = 0;
  let lengthDigit = 4;
  
  for (let i = 0; i < lengthDigit; i++) {
    const digit = Math.floor(Math.random() * maxDigit + 1)
    num.push(digit)
    sum += digit;
  }

  const totalDigit = Number(num.join(''))

  if (sum === 21) return totalDigit
  return SumNumMaxDigit(maxDigit);
}

console.log(SumNumMaxDigit(6));